src/App.tsx

```diff
@@ -14,11 +15,99 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  newList: string[],
+  click: boolean,
+  isReversed: boolean,
+  sortBy: string,
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    newList: goodsFromServer,
+    click: false,
+    isReversed: false,
+    sortBy: '',
+  };
+
+  sortByName = () => {
+    this.setState({ sortBy: 'name' });
+  };
+
+  sortByLength = () => {
+    this.setState({ sortBy: 'length' });
+  };
+
+  reset = () => {
+    this.setState({
+      isReversed: false,
+      sortBy: '',
+    });
+  };
+
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  render() {
+    const {
+      newList,
+      click,
+      isReversed,
+      sortBy,
+    } = this.state;
+
+    const copy = [...newList];
+
+    copy.sort((prev, next) => {
+      switch (sortBy) {
+        case 'name':
+          return prev.localeCompare(next);
+
+        case 'length':
+          return prev.length - next.length;
+
+        default:
+          return 0;
+      }
+    });
+
+    if (isReversed) {
+      copy.reverse();
+    }
+
+    return (
+      <div className="App">
+        <h1>Goods</h1>
+        {goodsFromServer.length}
```

Comment:
this looks weird when you don't know the meaning of this number ;)

<img width="121" alt="Screenshot 2022-04-21 at 13 44 34" src="https://user-images.githubusercontent.com/57598825/164441925-b496c1b8-fe51-4c27-a2e8-8054dc133972.png">